Add configuration option to suppress "unknown asset type" ERROR messages

UniFi Poller v2.21.0 logs ERROR messages for newer UniFi device types it doesn't recognize, generating significant log volume that impacts logging infrastructure performance (loki, alloy, etc).  Each message is several KB that's piped out to stdout.


```
2026/01/23 18:03:22 [ERROR] unknown asset type -....
```


It would be awesome to have something like `UP_POLLER_IGNORE_UNKNOWN_DEVICES` that would filter out these types of messages.  A better solution would be to allow people to control the `LOG_LEVEL` however since this is classified as and error it wouldn't really help here. (I think of these more as debug anyway since they really aren't actionable by the user/operator.)

Anyway, thank you for Unpoller!

--- a/examples/remote_api_example.conf
+++ b/examples/remote_api_example.conf
@@ -7,6 +7,10 @@
 [poller]
 debug = false
 quiet = false
+# log_unknown_types = false   # Set to true to log unknown device types as DEBUG messages.
+                              # By default, newer UniFi device types that aren't recognized
+                              # are silently ignored to reduce log volume. Enable this when
+                              # debugging or reporting new device types to developers.
 
 [unifi]
 remote = true

--- a/examples/remote_api_example.yaml
+++ b/examples/remote_api_example.yaml
@@ -7,6 +7,10 @@
 poller:
   debug: false
   quiet: false
+  # log_unknown_types: false   # Set to true to log unknown device types as DEBUG messages.
+                               # By default, newer UniFi device types that aren't recognized
+                               # are silently ignored to reduce log volume. Enable this when
+                               # debugging or reporting new device types to developers.
 
 unifi:
   # Enable remote API mode - automatically discovers all consoles

--- a/examples/up.conf.example
+++ b/examples/up.conf.example
@@ -11,6 +11,11 @@
   # Recommend enabling debug with this setting for better error logging.
   quiet = false
 
+  # Log unknown device types as DEBUG messages. By default (false), newer UniFi
+  # device types that aren't recognized are silently ignored to reduce log volume.
+  # Enable this when debugging or reporting new device types to developers.
+  # log_unknown_types = false
+
   # Load dynamic plugins. Advanced use; only sample mysql plugin provided by default.
   plugins = []
 

--- a/examples/up.json.example
+++ b/examples/up.json.example
@@ -2,7 +2,8 @@
   "poller": {
     "debug":   false,
     "quiet":   false,
-    "plugins": []
+    "plugins": [],
+    "log_unknown_types": false
   },
 
   "prometheus": {

--- a/examples/up.yaml.example
+++ b/examples/up.yaml.example
@@ -9,6 +9,10 @@ poller:
   debug: false
   quiet: false
   plugins: []
+  # log_unknown_types: false   # Set to true to log unknown device types as DEBUG messages.
+                               # By default, newer UniFi device types that aren't recognized
+                               # are silently ignored to reduce log volume. Enable this when
+                               # debugging or reporting new device types to developers.
 
 prometheus:
   disable:       false

--- a/pkg/influxunifi/influxdb.go
+++ b/pkg/influxunifi/influxdb.go
@@ -467,6 +467,8 @@ func (u *InfluxUnifi) switchExport(r report, v any) { //nolint:cyclop
 	case *unifi.SpeedTestResult:
 		u.batchSpeedTest(r, v)
 	default:
-		u.LogErrorf("invalid export type: %T", v)
+		if u.Collector.Poller().LogUnknownTypes {
+			u.LogDebugf("unknown export type: %T", v)
+		}
 	}
 }

--- a/pkg/poller/config.go
+++ b/pkg/poller/config.go
@@ -102,9 +102,10 @@ type Config struct {
 
 // Poller is the global config values.
 type Poller struct {
-	Plugins []string `json:"plugins" toml:"plugins" xml:"plugin"     yaml:"plugins"`
-	Debug   bool     `json:"debug"   toml:"debug"   xml:"debug,attr" yaml:"debug"`
-	Quiet   bool     `json:"quiet"   toml:"quiet"   xml:"quiet,attr" yaml:"quiet"`
+	Plugins         []string `json:"plugins"           toml:"plugins"           xml:"plugin"               yaml:"plugins"`
+	Debug           bool     `json:"debug"             toml:"debug"             xml:"debug,attr"           yaml:"debug"`
+	Quiet           bool     `json:"quiet"             toml:"quiet"             xml:"quiet,attr"           yaml:"quiet"`
+	LogUnknownTypes bool     `json:"log_unknown_types" toml:"log_unknown_types" xml:"log_unknown_types"    yaml:"log_unknown_types"`
 }
 
 // LoadPlugins reads-in dynamic shared libraries.

--- a/pkg/promunifi/collector.go
+++ b/pkg/promunifi/collector.go
@@ -452,6 +452,8 @@ func (u *promUnifi) switchExport(r report, v any) {
 	case *unifi.UsageByCountry:
 		u.exportCountryTraffic(r, v)
 	default:
-		u.LogErrorf("invalid type: %T", v)
+		if u.Collector.Poller().LogUnknownTypes {
+			u.LogDebugf("unknown type: %T", v)
+		}
 	}
 }

